### ISSUE TITLE
nuxt generate stalls on Generating pages

### ISSUE THREAD

**Issue by andrewharvey** [nuxt/nuxt.js#5546]
### Version

[2.6.2](https://github.com/nuxt.js/releases/tag/2.6.2) and 2.6.0 and later

### Reproduction link

sorry I don't have a repo link

### Steps to reproduce

With version 2.5.1 when I run nuxt generate with around 500 routes it finishes in about 21 minutes.

Upgrading to nuxt 2.6.2 with everything else the same nuxt generate gets stuck at Generating pages. My CI job terminates at 2hrs, it's possible the job might finish, but I don't know, that's a big jump from 21minutes in v2.5.1.

### What is expected ?

nuxt generate to finish in a reasonable time/not hang

### What is actually happening?

nuxt generate is either taking  over 4 times as long or getting stuck and never finishing

<div align="right"><sub><em>This bug report is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c9061">#c9061</a>)</em></sub></div>

**Comment by manniL** [comment]
might be related to #5324 and similar?

**Comment by andrewharvey** [comment]
I'm using node v10 LTS

**Comment by andrewharvey** [comment]
I'm still able to reproduce this issue with the latest release 2.6.3. Fixing to 2.5.1 it works.

**Comment by manniL** [comment]
hard to tell w/o a repo

**Comment by Youdaman** [comment]
For anyone else that comes across this, I had a similar issue where a previous version of nuxt generated numerous routes fine, but after an upgrade to the latest it would hang.

The problem seemed to be that I was fetching a big chunk of data via axios in the asyncData section of each page, and so instead I moved my data to a json file that I included by way of a module instead and that fixed the issue for me.

**Comment by Alfredo-Delgado** [comment]
I encountered this problem, too. Unfortunately, I also don't have a repo that I can readily share. :/

When `nuxt generate` hangs I have to kill (9) the process. I'm on node v8.15.0

Like with @andrewharvey, switching to 2.5.1 worked.

I also looked into where it seemed to be dying with @Youdaman's input in mind and I found a sloppy query in a work-in-progress page that was pulling down unnecessary data. Fixing that made the problem go away. However, It does seem like something regressed after 2.5.1.
